File: browser-use-python/examples/streaming.py
"""
Streaming -- monitor task progress in real time.
"""
import asyncio

from dotenv import load_dotenv
from browser_use_sdk import AsyncBrowserUse

load_dotenv()


async def main():
    client = AsyncBrowserUse()

    run = client.run(
        "Go to wikipedia.org and find the featured article of the day."
    )

    async for step in run:
        print(f"[Step {step.number}] {step.next_goal} -- {step.url}")

    print(f"\nOutput: {run.result.output}")


asyncio.run(main())

File: browser-use-node/examples/streaming.ts
/**
 * Streaming — monitor task progress in real time.
 */
import "dotenv/config";
import { BrowserUse } from "browser-use-sdk";

async function main() {
  const client = new BrowserUse();

  for await (const step of client.run(
    "Go to wikipedia.org and find the featured article of the day.",
  )) {
    console.log(`[${step.number}] ${step.nextGoal} — ${step.url}`);
  }

  const task = client.run("Summarize the wikipedia.org main page.");
  for await (const step of task) {
    console.log(`[${step.number}] ${step.nextGoal} — ${step.url}`);
  }
  console.log("\nOutput:", task.result!.output);
}

main();
